Source/ActionController.swift

```diff
@@ -584,40 +654,44 @@ open class ActionController<ActionViewType: UICollectionViewCell, ActionDataType
         default:
             prevPath = nil
         }
-        
+
         guard let unwrappedPrevPath = prevPath else { return nil }
-        
+
         return collectionView.cellForItem(at: unwrappedPrevPath) as? ActionViewType
     }
 
     func hasHeader() -> Bool {
         return headerData != nil && headerSpec != nil
     }
-    
+
+    var hasCancel: Bool {
+        return cancelSpec != nil  //settings.cancel.hasCancel
+    }
+
+    var cancelIndexPath: IndexPath? {
+        guard hasCancel else { return nil }
+        return IndexPath(item: _sections.last?.actions.count ?? 0, section: numberOfSections() - 1)
+    }
+
+
     fileprivate func numberOfActions() -> Int {
         return _sections.flatMap({ $0.actions }).count
     }
-        
+
     fileprivate func numberOfSections() -> Int {
         return hasHeader() ? _sections.count + 1 : _sections.count
     }
-    
+
     fileprivate func actionSectionIndexFor(_ section: Int) -> Int {
         return hasHeader() ? section - 1 : section
     }
 
-    fileprivate func setUpContentInsetForHeight(_ height: CGFloat) {
+    open func setUpContentInsetForHeight(_ height: CGFloat) {
         let currentInset = collectionView.contentInset
-        let bottomInset = settings.cancelView.showCancel ? settings.cancelView.height : currentInset.bottom
         var topInset = height - contentHeight
-        
-        if settings.cancelView.showCancel {
-            topInset -= settings.cancelView.height
-        }
-        
         topInset = max(topInset, max(30, height - contentHeight))
-        
-        collectionView.contentInset = UIEdgeInsets(top: topInset, left: currentInset.left, bottom: bottomInset, right: currentInset.right)
+
+        collectionView.contentInset = UIEdgeInsets(top: topInset, left: currentInset.left, bottom: 0 /*bottomInset*/, right: currentInset.right)
```

Comment:
Sorry for the hardcore. 
Now the bottomInset is equal to currentInset.bottom (0), because the cancelView doesn't exist anymore.  

I will change this to: let bottomInset = currentInset.bottom